joern-cli/frontends/jimple2cpg/src/main/scala/io/joern/jimple2cpg/astcreation/AstCreator.scala

```diff
@@ -49,9 +49,11 @@ class AstCreator(
     val astRoot = astForCompilationUnit(cls)
     storeInDiffGraph(astRoot, diffGraph)
 
-    val fileNode = NewFile().name(filename.replace(".class", ".java")).order(0)
-    fileContent.foreach(fileNode.content(_))
-    diffGraph.addNode(fileNode)
+    if (fileContent.isDefined) {
+      val fileNode = NewFile().name(filename.replace(".class", ".java")).order(0)
```

Comment:
Given that this is a bytecode frontend, the source file will always be the `.class` file
```suggestion
      val fileNode = NewFile().name(filename)).order(0)
```

Reply:
This also ensures that other nodes that set the `filename` property are linked to this node

Reply:
So the name on this fileNode should be the `.class` instead of the `.java`?

Reply:
To make sure this works, enable file content on the configs for File and Method tests and see if they still pass

Reply:
> So the name on this fileNode should be the `.class` instead of the `.java`?

Yes

Reply:
Change made, confirmed still working for File and Method tests with file content enabled.

![image](https://github.com/joernio/joern/assets/11736140/07fb18b7-5223-4e62-aeb8-734efb469b4b)

![image](https://github.com/joernio/joern/assets/11736140/ea743dbc-e1c3-450e-b773-c24003eb021f)

Reply:
Great!